src/amo/components/CollectionManager/index.js

```diff
@@ -211,12 +213,24 @@ export class CollectionManagerBase extends React.Component<
       creating,
       currentUsername,
       errorHandler,
+      filters,
       i18n,
       isCollectionBeingModified,
+      showEditForm,
       siteLang,
     } = this.props;
     const { description, name, slug } = this.state;
 
+    if (!showEditForm) {
+      return (
+        <CollectionDetails
```

Comment:
Handling this logic here puts an undue burden on `CollectionManager`. The logic burden is already on `Collection` so why not continue handling it there? Unless I'm missing something, keeping it in `Collection` is as simple as:

```diff
diff --git a/src/amo/components/Collection/index.js b/src/amo/components/Collection/index.js
index ab3bb3b14..067d5e0b3 100644
--- a/src/amo/components/Collection/index.js
+++ b/src/amo/components/Collection/index.js
@@ -318,7 +318,8 @@ export class CollectionBase extends React.Component<InternalProps> {
       showEditButton,
     } = this.props;
 
-    if (creating || editing) {
+    const managingCollection = creating || editing;
+    if (managingCollection && hasEditPermission) {
       return (
         <CollectionManager
           collection={collection}
@@ -333,7 +334,7 @@ export class CollectionBase extends React.Component<InternalProps> {
       <CollectionDetails
         collection={collection}
         filters={filters}
-        showEditButton={showEditButton}
+        showEditButton={showEditButton && !managingCollection}
       />
     );
   }
```